src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -28,11 +28,14 @@ public string GetString()
 
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
 
-            int idx = span.IndexOf(JsonConstants.BackSlash);
-            if (idx != -1)
+            if (_stringHasEscaping)
             {
+                int idx = span.IndexOf(JsonConstants.BackSlash);
```

Comment:
I wonder if we should store the `int idx` itself rather than `bool idx >= 0`

Reply:
It's currently possible, but I didn't want to tackle the complications of it in the reader (it's easy to say "no escaping needed" for chunked reading, but I wasn't sure where the index made sense).

It also takes a lot of the storage data, and I'm not sure it's worth that much space right now.  (e.g. maybe a better use of that space is an index into a secondary table where it's already been unescaped)